cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -524,6 +530,19 @@ private String generateFileSetCreateStatement(DatasetId datasetId, Dataset datas
     String databaseName = ExploreProperties.getExploreDatabaseName(properties);
     Map<String, String> tableProperties = FileSetProperties.getTableProperties(properties);
 
+    // if this dataset reuses an existing table, do not attempt to create it
+    if (FileSetProperties.isUseExisting(tableProperties)
+      || FileSetProperties.isPossessExisting(tableProperties)) {
+      try {
+        exploreService.getTableInfo(databaseName, tableName);
+        // table exists: do not attempt to create
+        return null;
+      } catch (Exception e) {
+        LOG.warn("Dataset '{}' is configured to use an existing explore table, but table '{}' does not " +
```

Comment:
:) probably true! I found a way to pass the isTruncating flag through the InternalEnableParams, without creating a new endpoint. Please see changes in latest commit